components/core/src/clp/Query.hpp

```diff
@@ -36,32 +35,36 @@ class QueryVar {
     /**
      * Removes segments from the given set that don't contain the given variable
      * @param segment_ids
+     * @param get_segments_containing_var_dict_id
      */
-    void remove_segments_that_dont_contain_dict_var(std::set<segment_id_t>& segment_ids) const;
+    void remove_segments_that_dont_contain_dict_var(
+            std::set<segment_id_t>& segment_ids,
+            std::function<std::set<segment_id_t> const&(variable_dictionary_id_t)>
+                    get_segments_containing_var_dict_id
+    ) const;
 
     bool is_precise_var() const { return m_is_precise_var; }
 
     bool is_dict_var() const { return m_is_dict_var; }
 
-    VariableDictionaryEntry const* get_var_dict_entry() const { return m_var_dict_entry; }
+    variable_dictionary_id_t get_var_dict_entry() const { return m_var_dict_entry; }
 
-    std::unordered_set<VariableDictionaryEntry const*> const&
-    get_possible_var_dict_entries() const {
+    std::unordered_set<variable_dictionary_id_t> const& get_possible_var_dict_entries() const {
         return m_possible_var_dict_entries;
     }
 
 private:
     // Variables
-    bool m_is_precise_var;
-    bool m_is_dict_var;
+    bool m_is_precise_var{};
+    bool m_is_dict_var{};
 
-    encoded_variable_t m_precise_var;
+    encoded_variable_t m_precise_var{};
     // Only used if the precise variable is a dictionary variable
-    VariableDictionaryEntry const* m_var_dict_entry;
+    variable_dictionary_id_t m_var_dict_entry{};
```

Comment:
since we don't store pointer to entries anymore, how about renaming them tom something like m_var_dict_id